🐛  CMake Error: REACT_NATIVE_UTILS_LIB set to NOTFOUND

### What were you trying to do?

Build my RN app on the Android emulator to prototype with frame processors.

**Note**: I checked the Expo box below. I'm running Expo bare, having ejected from Expo managed awhile back.

### What happened instead?

Gradle build failed with a CMake error when trying to finish tasks for VisionCamera.

### Relevant log output

```shell
/Users/dchiu/Developer/midana/client/node_modules/react-native-vision-camera/android/CMakeLists.txt : C/C++ debug|x86 : CMake Error: The following variables are used in this project, but they are set to NOTFOUND.
Please set them or make sure they are set and tested correctly in the CMake files:
REACT_NATIVE_UTILS_LIB
    linked by target "VisionCamera" in directory /Users/dchiu/Developer/midana/client/node_modules/react-native-vision-camera/android
```


### Device

Emulated Pixel 4 running API 30

### VisionCamera Version

2.4.2-beta.9

### Additional information

- [X] I am using Expo
- [X] I have read the [Troubleshooting Guide](https://mrousavy.github.io/react-native-vision-camera/docs/guides/troubleshooting)
- [X] I agree to follow this project's [Code of Conduct](https://github.com/mrousavy/react-native-vision-camera/blob/main/CODE_OF_CONDUCT.md)
- [X] I searched for [similar issues in this repository](https://github.com/mrousavy/react-native-vision-camera/issues) and found none.

--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -54,12 +54,32 @@ target_include_directories(
 # find libraries
 
 file (GLOB LIBRN_DIR "${BUILD_DIR}/react-native-0*/jni/${ANDROID_ABI}")
-file (GLOB LIBJSC_DIR "${BUILD_DIR}/android-jsc*.aar/jni/${ANDROID_ABI}")
-file (GLOB LIBHERMES_DIR "${BUILD_DIR}/third-party-ndk/hermes/jni/${ANDROID_ABI}")
 
 if(${FOR_HERMES})
+else()
+endif()
+
+if(${FOR_HERMES})
+    file (GLOB LIBHERMES_DIR "${BUILD_DIR}/third-party-ndk/hermes/jni/${ANDROID_ABI}")
+    # Use Hermes
+    find_library(
+            JS_ENGINE_LIB
+            hermes
+            PATHS ${LIBHERMES_DIR}
+            NO_CMAKE_FIND_ROOT_PATH
+    )
+    # Use Reanimated Hermes
     file (GLOB LIBREANIMATED_DIR "${BUILD_DIR}/react-native-reanimated-*-hermes.aar/jni/${ANDROID_ABI}")
 else()
+    file (GLOB LIBJSC_DIR "${BUILD_DIR}/android-jsc*.aar/jni/${ANDROID_ABI}")
+    # Use JSC
+    find_library(
+            JS_ENGINE_LIB
+            jscexecutor
+            PATHS ${LIBRN_DIR}
+            NO_CMAKE_FIND_ROOT_PATH
+    )
+    # Use Reanimated JSC
     file (GLOB LIBREANIMATED_DIR "${BUILD_DIR}/react-native-reanimated-*-jsc.aar/jni/${ANDROID_ABI}")
 endif()
 
@@ -95,43 +115,16 @@ find_library(
         NO_CMAKE_FIND_ROOT_PATH
 )
 
-find_library(
-        HERMES_LIB
-        hermes
-        PATHS ${LIBHERMES_DIR}
-        NO_CMAKE_FIND_ROOT_PATH
-)
-find_library(
-        JSEXECUTOR_LIB
-        jscexecutor
-        PATHS ${LIBRN_DIR}
-        NO_CMAKE_FIND_ROOT_PATH
-)
-
 # linking
 
 message(WARNING "VisionCamera linking: FOR_HERMES=${FOR_HERMES}")
-
-if(${FOR_HERMES})
-    target_link_libraries(
-            ${PACKAGE_NAME}
-            ${LOG_LIB}
-            ${HERMES_LIB}
-            ${REANIMATED_LIB}
-            ${REACT_NATIVE_JNI_LIB}
-            ${FBJNI_LIB}
-            ${FOLLY_JSON_LIB}
-            android
-    )
-else()
-    target_link_libraries(
-            ${PACKAGE_NAME}
-            ${LOG_LIB}
-            ${JSEXECUTOR_LIB}
-            ${REANIMATED_LIB}
-            ${REACT_NATIVE_JNI_LIB}
-            ${FBJNI_LIB}
-            ${FOLLY_JSON_LIB}
-            android
-    )
-endif()
+target_link_libraries(
+        ${PACKAGE_NAME}
+        ${LOG_LIB}
+        ${JS_ENGINE_LIB} # <-- Hermes or JSC
+        ${REANIMATED_LIB}
+        ${REACT_NATIVE_JNI_LIB}
+        ${FBJNI_LIB}
+        ${FOLLY_JSON_LIB}
+        android
+)

--- a/android/CMakeLists.txt
+++ b/android/CMakeLists.txt
@@ -96,13 +96,6 @@ find_library(
 )
 
 find_library(
-        REACT_NATIVE_UTILS_LIB
-        reactnativeutilsjni
-        PATHS ${LIBRN_DIR}
-        NO_CMAKE_FIND_ROOT_PATH
-)
-
-find_library(
         HERMES_LIB
         hermes
         PATHS ${LIBHERMES_DIR}
@@ -126,7 +119,6 @@ if(${FOR_HERMES})
             ${HERMES_LIB}
             ${REANIMATED_LIB}
             ${REACT_NATIVE_JNI_LIB}
-            ${REACT_NATIVE_UTILS_LIB}
             ${FBJNI_LIB}
             ${FOLLY_JSON_LIB}
             android
@@ -138,7 +130,6 @@ else()
             ${JSEXECUTOR_LIB}
             ${REANIMATED_LIB}
             ${REACT_NATIVE_JNI_LIB}
-            ${REACT_NATIVE_UTILS_LIB}
             ${FBJNI_LIB}
             ${FOLLY_JSON_LIB}
             android

